alacritty/src/display.rs

```diff
@@ -0,0 +1,461 @@
+            padding_x = dynamic_padding(padding_x, viewport_size.width as f32, cell_width);
+            padding_y = dynamic_padding(padding_y, viewport_size.height as f32, cell_height);
+        }
+
+        padding_x = padding_x.floor();
+        padding_y = padding_y.floor();
+
+        info!("Cell Size: {} x {}", cell_width, cell_height);
+        info!("Padding: {} x {}", padding_x, padding_y);
+
+        let size_info = SizeInfo {
+            dpr,
+            width: viewport_size.width as f32,
+            height: viewport_size.height as f32,
+            cell_width: cell_width as f32,
+            cell_height: cell_height as f32,
+            padding_x: padding_x as f32,
+            padding_y: padding_y as f32,
+        };
+
+        // Update OpenGL projection
+        renderer.resize(&size_info);
+
+        // Clear screen
+        let background_color = config.colors.primary.background;
+        renderer.with_api(&config, &size_info, |api| {
+            api.clear(background_color);
+        });
+
+        // We should call `clear` when window is offscreen, so when `window.show()` happens it
+        // would be with background color instead of uninitialized surface.
+        window.swap_buffers();
+
+        window.set_visible(true);
+
+        // Set window position
+        //
+        // TODO: replace `set_position` with `with_position` once available
+        // Upstream issue: https://github.com/tomaka/winit/issues/806
+        if let Some(position) = config.window.position {
+            let physical = PhysicalPosition::from((position.x, position.y));
+            let logical = physical.to_logical(window.hidpi_factor());
+            window.set_outer_position(logical);
+        }
+
+        #[allow(clippy::single_match)]
+        match config.window.startup_mode() {
+            StartupMode::Fullscreen => window.set_fullscreen(true),
+            #[cfg(target_os = "macos")]
+            StartupMode::SimpleFullscreen => window.set_simple_fullscreen(true),
+            #[cfg(not(any(target_os = "macos", windows)))]
+            StartupMode::Maximized => window.set_maximized(true),
+            _ => (),
+        }
+
+        Ok(Display { window, renderer, glyph_cache, meter: Meter::new(), size_info })
+    }
+
+    fn new_glyph_cache(
+        dpr: f64,
+        renderer: &mut QuadRenderer,
+        config: &Config,
+    ) -> Result<(GlyphCache, f32, f32), Error> {
+        let font = config.font.clone();
+        let rasterizer = font::Rasterizer::new(dpr as f32, config.font.use_thin_strokes())?;
+
+        // Initialize glyph cache
+        let glyph_cache = {
+            info!("Initializing glyph cache...");
+            let init_start = Instant::now();
+
+            let cache =
+                renderer.with_loader(|mut api| GlyphCache::new(rasterizer, &font, &mut api))?;
+
+            let stop = init_start.elapsed();
+            let stop_f = stop.as_secs() as f64 + f64::from(stop.subsec_nanos()) / 1_000_000_000f64;
```

Comment:
I think `stop_f` should be replaced with just `stop`, I don't really like that explicit type thing